workspace/solution.go

```diff
@@ -87,3 +95,23 @@ func (s *Solution) PathToParent() string {
 	}
 	return filepath.Join(dir, s.Track)
 }
+
+func createIgnoreSubdir(path string) (string, error) {
+	path = filepath.Join(path, IgnoreSubdir)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if err := os.Mkdir(path, os.FileMode(0755)); err != nil {
+			return "", fmt.Errorf("failed to create directory: %s", path)
+		}
+	}
+	return path, nil
+}
+
+func migrateLegacySolutionFile(legacySolutionPath string, solutionPath string) error {
+	if _, err := createIgnoreSubdir(filepath.Dir(legacySolutionPath)); err != nil {
+		return err
+	}
+	if err := os.Rename(legacySolutionPath, solutionPath); err != nil {
```

Comment:
What happens if there is no solution file old or legacy? 

Should we be checking if the file exists before trying to migrate it? With that said, I think we should print to Stderr that the solution is being migrated so that the user is made aware that a file is being changed for them. 

This could prove helpful for those users who have their files under version control.

Reply:
> What happens if there is no solution file old or legacy?

Then errMissingMetadata gets handled (`migrateLegacySolutionFile` never gets called).

> Should we be checking if the file exists before trying to migrate it?

In this implementation, `migrateLegacySolutionFile` only gets called when a legacy solution (`$EXERCISE_ROOT/.solution.json`) is found (this is checked before checking for `$EXERCISE_ROOT/.exercism/solution.json`). It is potentially dangerous and poor defensive programming practice not to check this inside the actual function though. Good call.

In the case that both legacy and modern solution files exist (due to user manually moving files around) the legacy would overwrite the modern solution. It's not apparent to me that this would be undesirable but I don't have a ton of experience with the product and it's quite possible I'm ignorant here.

In the case where both legacy and modern solutions exist, what should be done? Which solution file is more likely to be correct (I'd bet the legacy because the only sensible way I imagine this case occuring is if a user downloads a new exercise and copies the contents of a legacy solution over)? I see the options as:

1. Don't migrate, print Stderr, no error => user should be able to successfully submit (but has a junk legacy solution file left in the workspace)
2. Same as 1. but delete the legacy metadata file for the user
2. Same as 1. but replace the modern metadata file with the legacy
4. Don't migrate, print Stderr, return error => user will get the missing metadata error (this seems confusing and bad)

What to print to Stderr? Perhaps: "Attempted to migrated solution metadata to %s but file already exists."


> With that said, I think we should print to Stderr that the solution is being migrated so that the user is made aware that a file is being changed for them.

Good call, I'll add it.